### DISCUSSION TITLE
Unable to get RLS to work locally in common use case?

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [supabase/supabase#19383] [Questions] [answered]
I'd actually posted this question over in [r/supabase](https://www.reddit.com/r/Supabase/comments/188no96/unable_to_get_rls_to_work_in_simple_case/), and while there were some good responses, nothing seems to work still.

To recap and to provide some context, I have a locally hosted Supabase instance, as well as a separated frontend and backend. The frontend handles auth with the SB helpers, and the backend returns more complex GraphQL queries and mutations. 

I have a public.users table that automatically inserts records via a trigger and function. The public.users schema has an id column that directly references the auth.users id. All of this works well so far.

I can make a request (without RLS enabled), and the data returns. I then enable RLS, and added the following policy:
`CREATE POLICY "Can only view own user record" ON public.users FOR SELECT USING (auth.uid() = id);`

Once I add the policy and enable RLS though, the query returns null for error and data.

```
    const { data: user, error } = await supabase
      .from('users')
      .select('*')
      .eq('id', id)
      .maybeSingle();
    console.log({ user });
    console.log({ error });
```

I know that the user is authenticated, that the id being passed is correct, and that simplifying the select to just .select('*') doesn't fix it. 
I've also tried updating the policy to public, authenticated, and anon but none of those fix the issue. At this point I feel like it's something dumb I'm overlooking. 

The ONE clue I've found is that by changing the USING expression to just `true`, that allows the query to resolve with the data I'm looking for. As soon as I change it back to `USING (auth.uid() = id);`, it fails. 

Is it an issue with the two separate codebases? Or maybe an issue with RLS working on a locally hosted instance?  Does anyone have an idea what it could be?

**Comment by GaryAustin1** [comment]
There is not likely a Postgres issue in local.  
have you tried Anon for the TO and true? If that works then you don’t really have a session when you make the call.  Is your column name lower case id?   
If you are using Next.js it is causing users fits when debugging RLS because it caches by URLs in many cases.

**Reply by jimmiejackson414** [reply]
If i set it to true, it does work. It seems to be an issue with the 'auth.uid()' helper. I know a couple of years ago i used to manually 'setAuth' but i believe that's been deprecated now? I'm thinking I'll need to do something similar since my back and front end are separate

**Reply by GaryAustin1** [reply]
If anon worked then that means you don't have a session when you make the call so auth.uid() will be null.   If authenticated role and true works that would be different.   The Authorization header has to be 'bearer userjwt'

**Reply by jimmiejackson414** [reply]
Authenticated + true expression does not work (so that's good). What do you think about - upon user login from the frontend - sending a request along with the token to my separate backend to `setSession()`?

**Reply by GaryAustin1** [reply]
So authenticated and true means you do not have a signed in user, or you do not have a session in the client at the point you make the call.
You will need to figure out why.  You might ask a new question if you can't figure that out as it is very dependent on your framework and how you are doing create client.  This is especially true if server code is involved.

**Reply by jimmiejackson414** [reply, marked as answer]
I appreciate the rubber ducking session! I _believe_ I just got it fixed by doing what I'd just mentioned: as the frontend logs in the user, I send a secondary request to the backend to `setSession`, passing along the token. All subsequent queries work off that session now, and RLS appears to be working correctly!
